app/components/app-nav-primary.hbs

```diff
@@ -0,0 +1,44 @@
+{{#if this.show}}
+  <div
+    {{focus-trap
+      isActive=(media 'isMobile')
+      focusTrapOptions=(hash
+        onDeactivate=(action @toggleNavPrimary)
+      )
+    }}>
+    <nav class="app-nav-primary {{if @navPrimaryIsOpen 'app-nav-primary--open'}}">
+      <button
+        class="app-nav-primary__mobile-nav-toggle"
+        aria-label="Hide Site Navigation"
+        {{on "click" @toggleNavPrimary}}>
+        <svg aria-hidden="true" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" stroke="currentColor">
+          <g>
+            <line x1="3" y1="12" x2="21" y2="12"></line>
+            <line x1="3" y1="6" x2="21" y2="6"></line>
+            <line x1="3" y1="18" x2="21" y2="18"></line>
+          </g>
+        </svg>
+      </button>
+
+      <em>TODO: Make into real links...</em>
+
+      <ul class="app-nav-primary__list">
+        {{#unless isLoading}}
+          {{#each this.contents as | content |}}
+            <li> {{!-- Convert to LinkTo's when we have content page routing done --}}
+              {{content.title}}
+            </li>
+          {{/each}}
+        {{/unless}}
+      </ul>
+    </nav>
+
+    {{#if @navPrimaryIsOpen}}
+      <div
+        role="button"
```

Comment:
This feels kind of murky from and a11y perspective to use `role="button"` here. I guess I don't know how much of the page this div would encompass. I would think that an `aria-label` would be in order, too.  It might be worth test with a SR if there's the option.

Reply:
I'm a _little_ unsure as well. The `role="button"` I'm ashamed to say was to appease the linter. However, in terms of accessibility perhaps it's best to hide this button altogether from screen-readers. The toggle button in the header has screen-reader friendly text with a clear action and makes way more sense than this div, which is just a way of capturing a "click-elsewhere" style thing.

Open to suggestions, I'm leaning in the direction of hiding this clickable div from screen-readers in favor of the toggle button.

Or another idea is to add a click event listener that truly listens for a click-elsewhere style event and we let the background fade be purely a CSS thing.

Reply:
I've implemented a click-elsewhere ember modifier that handles this instead!